Unify post components

nevyangelova opened the pull request:
#### Summary
Opening WIP PR for unifying the post components in case anyone wanted to take a look.  The idea behind this PR is to have one reusable component for root post, rhs root post, comment and search result item.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43162

#### Release Note

```release-note
NONE
```

nevyangelova commented:
Lets give it the first dev review rounds to ensure the approach is acceptable for everyone

M-ZubairAhmed commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,112 @@
+    const showActionsMenuPulsatingDot = showActionsDropdownPulsatingDot(state);
+
+    let emojis: Emoji[] = [];
+    const oneClickReactionsEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.ONE_CLICK_REACTIONS_ENABLED, Preferences.ONE_CLICK_REACTIONS_ENABLED_DEFAULT) === 'true';
+    if (oneClickReactionsEnabled) {
+        emojis = getOneClickReactionEmojis(state);
+    }
```
may be this can go its own selector, this has been used more than 2, now 3 times

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,332 @@
+        if (a11yActive) {
+            postRef.current?.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
+        }
+    }, [postRef.current]);
```
i dont think ref can be passed as dependency, since mutating them wont trigger render

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,333 @@
+        }
+    };
+
+    const handleFileDropdownOpened = (open: boolean) => setFileDropdownOpened(open);
```
These need to be wrapped in `useCallback` since they're passed into other components

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,333 @@
+        }
+    }, []);
+
+    useEffect(() => {
```
I think you might be using useEffect here incorrectly since these are triggering when `hover` changes, not when the component is unmounted

M-ZubairAhmed replied:
yes ☝🏽

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,333 @@
+        }
+    };
+
+    const setsHover = (e: MouseEvent<HTMLDivElement>) => {
```
Is there a more descriptive name that we can use for this? Maybe `handleMouseOver` and `handleMouseLeave` to match the corresponding event handlers?

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,333 @@
+        document.removeEventListener('keyup', handleAlt);
+    };
+
+    const handleA11yActivateEvent = () => setA11y(true);
+
+    const handleA11yDeactivateEvent = () => setA11y(false);
```
These can be moved into the `useEffect` to keep them all together

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,333 @@
+        }
+    }, [hover]);
+
+    const getClassName = (post: Post, isSystemMessage: boolean, isMeMessage: boolean) => {
```
Minor thing, perhaps we could move the call to `isMeMessage` into here instead of calling it outside of this and passing it in? If we inline it in the call to `classNames`, we can get rid of the naming conflict as well

hmhealey commented:
Could you also remove the old post components as well? Removing them early means that this PR will conflict with any changes on master that might be made to the post components sooner so that we don't end up accidentally reverting those changes when this is merged

ogi-m commented:
@nevyangelova Here's a list of issues I'm seeing:

1.  Hover highlight doesn't extent to do the date divider
![Off-Topic - pr11127 Mattermost - Google Chrome 10_20_2022 9_38_37 PM](https://user-images.githubusercontent.com/15314183/197167826-41e1c300-b761-42fe-97d1-d1596d9faa64.png)

2. Jump to post doesn't highlight the post

https://user-images.githubusercontent.com/15314183/197168471-bc99e209-6a60-41f7-85a1-520c448808ee.mp4

3. Mark as unread is not working

4. Post menus don't want to go away 😄 

https://user-images.githubusercontent.com/15314183/197173367-434ffcdf-2a7a-44a8-98a1-cdabadcc0dba.mp4

5. Search results post menu "Jump" button alignment is off (font looks too big):
![image](https://user-images.githubusercontent.com/15314183/197172119-4096be4c-3739-4ff0-bd11-611ce42810f6.png)

6.  Error when clicking on "Jump" from search results
![image](https://user-images.githubusercontent.com/15314183/197174724-184e5f44-1c92-4e93-a3f8-02e6a41c7ba4.png)

7. "Jump" should be visible on search results without hovering
master/PR:
![image](https://user-images.githubusercontent.com/15314183/197175766-17a785c8-ef37-4db7-aaa9-1c48ba9c55ca.png) 
![image](https://user-images.githubusercontent.com/15314183/197175825-98f5fb93-9cc0-497c-9cd0-cd0298950f70.png)

8. Thread footer shown in Threads view:
![image](https://user-images.githubusercontent.com/15314183/197173838-8bf63bde-9f72-47aa-9687-caf5b9c8c878.png)

9. Tooltip for Message actions jumps to the top left on hover

https://user-images.githubusercontent.com/15314183/197181581-056ff6a7-b719-4ede-bbeb-6e28d16372ce.mp4

nevyangelova commented:
@hmhealey @M-ZubairAhmed some things have changed due to uncaught bugs can you take a second look

nevyangelova commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/257066

hmhealey commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,228 @@
+function isConsecutivePost(state: GlobalState, ownProps: OwnProps) {
+    let post;
+    if (ownProps.postId) {
+        post = getPost(state, ownProps?.postId);
```
Nit, but this doesn't need `?.`
```suggestion
        post = getPost(state, ownProps.postId);
```

hmhealey commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,228 @@
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    let post;
+    if (ownProps.postId) {
+        post = getPost(state, ownProps?.postId);
```
Same here
```suggestion
        post = getPost(state, ownProps.postId);
```

hmhealey commented on components/new_post/post_options.tsx:
```diff
@@ -0,0 +1,279 @@
+    isSearchResultsItem?: boolean;
+    canReply?: boolean;
+    replyCount?: number;
+    location: 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT' | string;
```
Are we sure this needs to accept an arbitrary string as an option?

hmhealey replied:
Oh, I see that we're passing `Locations.SEARCH` into this. In that case, we can replace these types with `keyof typeof Locations` to avoid repeating ourselves
```suggestion
    location: keyof typeof Locations;
```

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,494 @@
+    matches?: string[];
+    term?: string;
+    isMentionSearch?: boolean;
+    location: 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT' | string;
```
Ditto here
```suggestion
    location: keyof typeof Locations;
```

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,494 @@
+    isMentionSearch?: boolean;
+    location: 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT' | string;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
```
And maybe here?

hmhealey commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,494 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {MouseEvent, useCallback, useEffect, useRef, useState} from 'react';
```
Also, could you merge in the latest master and re-run ESLint? It's going to complain about there not being a space with the change I made

ogi-m commented:
@nevyangelova I'm no longer seeing issues 1-7, but I can still repro issue 8 & 9.

And two newly found issues:
1. The highlight issue I noted is now fixed but now it doesn't extend to the date divider if it's below:
**Expected**:
![image](https://user-images.githubusercontent.com/15314183/200052570-5f7e21fd-3dde-415a-8c5a-07cc4a77f311.png)
**Observed**:
![image](https://user-images.githubusercontent.com/15314183/200052821-9a37a34f-304d-4798-8151-709fd44a88ce.png)

2. "Update your status" is not shown on posts for new users:
**Expected**:
![image](https://user-images.githubusercontent.com/15314183/200053324-19fefb17-6dae-40b9-b9b7-dab1e79bbe5b.png)
**Observed**:
![image](https://user-images.githubusercontent.com/15314183/200053360-ddbd6761-75c3-4455-8a61-d5ca128539b9.png)

nevyangelova commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/261876

jgilliam17 commented:
Thanks for running e2es @nevyangelova 
Significant number of tests failed 500+. See full [report here](https://mattermost-cypress-report.s3.amazonaws.com/261876-b53c691-onprem-ent-webapp-pr-11127-mm-ee-test:b53c691/mochawesome.html)

jgilliam17 commented:
Manually tested and seeing few issues, will continue testing tomorrow 
1. Search results replies have non-CRT formatting, despite CRT enabled 
<img width="1217" alt="Screen Shot 2022-11-14 at 7 31 47 PM" src="https://user-images.githubusercontent.com/52937121/201797143-7b055728-ec15-42f9-9d19-8a4ca47736d3.png">

2. "Click to open threads" user setting is ON, but click on thread is not opening post in RHS

3. Jump is not visible on the RHS without post hover

4. Remove Pinned post highlighting of the pinned and saved posts while displayed on the RHS

5. Remove Pinned label and icon from all posts on RHS when viewing Pinned posts list. Continue to display saved labels and icons on the saved posts included on the pinned posts list. e.g. if the post is both saved and pinned, only display saved label when viewing the post on the pinned RHS, hide pinned icon and label

6. Remove Saved label and icon from all posts on RHS when viewing Saved posts list. Continue to display pinned label and icon on the pinned posts included on the saved posts list. e.g. if the post is both saved and pinned, only display pinned label and icon when viewing the post on the saved posts RHS, hide saved icon and label

7. Remove post menu reply arrow when the post footer has the number of the replies link 
<img width="403" alt="Screen Shot 2022-11-14 at 7 49 51 PM" src="https://user-images.githubusercontent.com/52937121/201802406-c1a895ba-539d-4392-8479-ec3c3772fb7d.png">

M-ZubairAhmed commented on components/threading/virtualized_thread_viewer/reply/reply.tsx:
```diff
@@ -21,35 +18,28 @@ type Props = {
     previousPostId: string;
     teamId: string;
     timestampProps?: Partial<TimestampProps>;
+    id?: string;
```
nit : Assuming this is post id, `Post['id]`

M-ZubairAhmed commented on components/post_view/post_list_row/post_list_row.tsx:
```diff
@@ -28,6 +29,8 @@ export type PostListRowProps = {
     loadNewerPosts: () => void;
     togglePostMenu: (opened: boolean) => void;
+    post: Post;
+    currentUserId: string;
```
nit `UserProfile['id']`

M-ZubairAhmed commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,228 @@
+
+function removePostAndCloseRHS(post: ExtendedPost) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState() as GlobalState;
```
why did we had to implicitly type it to GlobalState?

nevyangelova replied:
this was a copy paste from another index file so I didn't test it, checking now, we have two types of global state, one from mm-redux and another specified in types/store. I guess we need a separate ticket to unify them

M-ZubairAhmed replied:
No actually we dont have to unify them since one of them is in regards to webapp specifically eg has plugin etc.

M-ZubairAhmed commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,228 @@
+    };
+}
+
+function mapDispatchToProps(dispatch: Dispatch<AnyAction>) {
+    return {
+        actions: bindActionCreators({
+            markPostAsUnread,
+            emitShortcutReactToLastPostFrom,
+            setActionsMenuInitialisationState,
+            selectPost,
+            setRhsExpanded,
+            removePost: removePostAndCloseRHS,
+            closeRightHandSide,
+            selectPostCard: selectPostCardFromRightHandSideSearch,
+        }, dispatch),
+    };
+}
```
something we discussed in web guild. this can be a plain object

nevyangelova replied:
sorry I missed this, is there an example in the app that you can show me?

M-ZubairAhmed replied:
components/integrations/delete_integration_link/index.ts

M-ZubairAhmed commented on components/new_post/post_options.tsx:
```diff
@@ -0,0 +1,280 @@
+                type='button'
+                onClick={removePost}
+            >
+                {'×'}
```
can this is icon?

M-ZubairAhmed commented on components/new_post/post_options.tsx:
```diff
@@ -0,0 +1,280 @@
+        return (
+            <button
+                className='post__remove theme color--link style--none'
+                type='button'
```
we dont need type button for this

M-ZubairAhmed commented on components/new_post/post_options.tsx:
```diff
@@ -0,0 +1,280 @@
+
+    const isPostDeleted = post && post.state === Posts.POST_DELETED;
+    const isEphemeral = isPostEphemeral(post);
+    const isSystemMessage = PostUtils.isSystemMessage(post);
```
can we selectively import `isSystemMessage` ?

M-ZubairAhmed commented on components/new_post/post_options.tsx:
```diff
@@ -0,0 +1,280 @@
+    if (isEphemeral) {
+        options = (
+            <div className='col col__remove'>
+                {createRemovePostButton()}
```
we can have the component directly here. instead of a function

M-ZubairAhmed commented on components/new_post/user_profile.tsx:
```diff
@@ -0,0 +1,141 @@
+
+const PostUserProfile = (props: Props): JSX.Element | null => {
+    const {post, compactDisplay, isMobileView, isConsecutivePost, isBusy, enablePostUsernameOverride, isBot, isSystemMessage, colorizeUsernames} = props;
+    const fromAutoResponder = PostUtils.fromAutoResponder(post);
```
selectively import `fromAuto..`

M-ZubairAhmed commented on components/new_post/user_profile.tsx:
```diff
@@ -0,0 +1,141 @@
+        }
+    }
+
+    return (<div className='col col__name'>
+        {userProfile}
+        {botIndicator}
+        {customStatus}
+    </div>);
+};
```
i dont know why eslint is not complaining for this indents!

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,498 @@
+import PostUserProfile from './user_profile';
+import PostOptions from './post_options';
+
+export type Props = {
+    post: Post;
+    teamId: string;
```
this is redundant since we already defined OwnProps in index file. and the redux props can be taken from connector. Check out components/channel_notifications_modal/index.ts

nevyangelova replied:
but this way we are doing circular imports tho, is this what we want?

M-ZubairAhmed replied:
it wont be because we do import type only
```js
import type {Foo} from 'xx'
```
Which will be stripped out eventually

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,498 @@
+import ThreadFooter from 'components/threading/channel_threads/thread_footer';
+import PostBodyAdditionalContent from 'components/post_view/post_body_additional_content';
+import PostMessageContainer from 'components/post_view/post_message_view';
+import * as Utils from 'utils/utils';
```
may be selectively import ?

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,498 @@
+} from 'mattermost-redux/utils/post_utils';
+
+import Constants, {A11yCustomEventTypes, AppEvents, Locations} from 'utils/constants';
+import * as PostUtils from 'utils/post_utils';
```
may be selectively import ?

nevyangelova replied:
I did it for the ones that we import only one function but here we import 6, is that something we wanna create a convention for?

M-ZubairAhmed replied:
i dont think there is any convention. Should'nt we import only what we need.

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,498 @@
+    const [fadeOutHighlight, setFadeOutHighlight] = useState(false);
+    const [alt, setAlt] = useState(false);
+
+    if (props.shouldHighlight) {
+        setTimeout(() => {
+            setFadeOutHighlight(true);
+        }, Constants.PERMALINK_FADEOUT);
+    }
```
we are not clearing the previous set time out here. leads to leaks

nevyangelova replied:
nice catch!

nevyangelova replied:
nice catch!

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,498 @@
+    const handleMouseOver = (e: MouseEvent<HTMLDivElement>) => {
+        setHover(true);
+        setAlt(e.altKey);
+        addKeyboardListeners();
```
we are adding a listener everytime we mouseover? 😕

nevyangelova replied:
right, this is another thing that was copied over but perhaps i should move them to the useEffect

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,498 @@
+            return;
+        }
+        props.actions.selectPost(post);
+    }, []);
```
should we add post as dependency here

M-ZubairAhmed reviewed (commented):
Did a first past of the review. Thank you

M-ZubairAhmed commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,233 @@
+    const showActionsMenuPulsatingDot = showActionsDropdownPulsatingDot(state);
+
+    let emojis: Emoji[] = [];
+    const oneClickReactionsEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.ONE_CLICK_REACTIONS_ENABLED, Preferences.ONE_CLICK_REACTIONS_ENABLED_DEFAULT) === 'true';
```
```suggestion
    const oneClickReactionsEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.ONE_CLICK_REACTIONS_ENABLED, Preferences.ONE_CLICK_REACTIONS_ENABLED_DEFAULT) === 'true');
```

M-ZubairAhmed commented on components/new_post/index.tsx:
```diff
@@ -0,0 +1,233 @@
+        channelIsArchived: isArchivedChannel(channel),
+        isConsecutivePost: isConsecutivePost(state, ownProps),
+        previousPostIsComment,
+        isFlagged: get(state, Preferences.CATEGORY_FLAGGED_POST, post.id, null) != null,
```
0/5 `!== null`

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,500 @@
+        if (a11yActive) {
+            postRef.current?.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
+        }
+    }, [hover, postRef.current]);
```
ref should not be used as dependencies, because they are not tracked for updates.

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,500 @@
+        document.addEventListener('keyup', handleAlt);
+    };
+
+    const removeKeyboardListeners = () => {
```
are we removing this listeners on unmount somewhere?

nevyangelova replied:
AFAIK with useEffect you don't think of the component lifecycle, but rather manipulate based on prop change. So I have split the two useEffects by concern. But I am open to a suggestion on where to add the removal of the listeners too?

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,500 @@
+            return;
+        }
+        props.actions.selectPost(props.post);
+    }, [props.post]);
```
```suggestion
    }, [props.post.id]);
```
Depending on do you want to change this function is other properties of post change such as count etc. If not then above dependency will suffice

M-ZubairAhmed commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,500 @@
+        props.actions.selectPost(props.post);
+    }, [props.post]);
+
+    const {
+        post,
+        isPostBeingEdited,
+    } = props;
```
we have desctructured the props here but in below line we are using `props.post` We can stick to either of one

M-ZubairAhmed commented on components/new_post/post_options.tsx:
```diff
@@ -0,0 +1,273 @@
+                postId={post.id}
+                teamId={props.teamId}
+                emojis={props.recentEmojis}
+                getDotMenuRef={getDotMenuRef}
```
i guess this prop is not used?

nevyangelova commented:
@jgilliam17 thank you! Comments are resolved. I will open another PR for the e2e tests so its easier to review

nevyangelova commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/266422

hmhealey commented on components/new_post/user_profile.tsx:
```diff
@@ -0,0 +1,141 @@
+    post: Post;
+    compactDisplay?: boolean;
+    colorizeUsernames?: boolean;
+    isBusy?: boolean;
```
I'm looking at the existing code, and I don't think `isBusy` is actually used any more. I'm pretty sure it goes back to the old pre-Calls, pre-Zoom plugin video calling thing we had that never quite worked

hmhealey commented on components/new_post/user_profile.tsx:
```diff
@@ -0,0 +1,141 @@
+    isConsecutivePost?: boolean;
+    isBot: boolean;
+    isSystemMessage: boolean;
+    isPostBeingEdited?: boolean;
```
This isn't used anywhere in here

hmhealey commented on components/new_post/user_profile.tsx:
```diff
@@ -0,0 +1,141 @@
+                userId={post.user_id}
+                channelId={post.channel_id}
+                isBusy={isBusy}
+                isRHS={true}
```
This still needs to be passed in unfortunately. Even though we aren't rendering the post differently in the RHS, this is still needed for a few things

nevyangelova replied:
fixing this to not be hardcoded in the e2e PR here
https://github.com/mattermost/mattermost-webapp/pull/11693

hmhealey commented on components/post_view/post_list_row/post_list_row.tsx:
```diff
@@ -173,8 +177,9 @@ export default class PostListRow extends React.PureComponent<PostListRowProps> {
-                postId={listId}
+            <PostComponent
+                post={this.props.post}
+                location={Locations.CENTER}
```
Is `PostListRow` used in the RHS at all, or does that use `ThreadViewerRow` nowadays?

nevyangelova replied:
You are right, this might be one of the components that's removed in the removal PR :P

hmhealey reviewed (changes requested):
Gave this another look through, and there's a few things I noticed, but everything else looked good.

Have the things that Zubair mentioned been addressed?

nevyangelova commented:
> Have the things that Zubair mentioned been addressed?

they have but he can take another look to verify

hmhealey reviewed (approved):
My stuff's been addressed either here or in that other PR. I'll do another quick pass after the E2E tests are done in case they uncover some other bugs, but this should be mostly good to go

hmhealey commented:
@M-ZubairAhmed Could you take another look at this to see if your feedback has been addressed as well?

M-ZubairAhmed commented on components/new_post/user_profile.tsx:
```diff
@@ -0,0 +1,136 @@
+        }
+    }
+
+    return (<div className='col col__name'>
+        {userProfile}
+        {botIndicator}
+        {customStatus}
+    </div>);
```
```suggestion
    return (
    <div className='col col__name'>
        {userProfile}
        {botIndicator}
        {customStatus}
    </div>
    );
```

M-ZubairAhmed reviewed (approved):
There are couple of suggestion which are missed. I have marked them as resolved and vise versa. Approving for now pending resolution to them

nevyangelova commented:
Apologies @M-ZubairAhmed i will go through the previous comments more thoroughly today

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/285971) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/285971)

koox00 commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,577 @@
+                                handleFileDropdownOpened={handleFileDropdownOpened}
+                            />
+                            }
+                            <ReactionList
+                                post={post}
+                            />
```
We should add post Acknowledgements in the same line as Reactions

koox00 commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,577 @@
+                                            aria-hidden='true'
+                                        />
+                                    </button>
+                                </OverlayTrigger>
```
The post's priority label should be displayed after the timestamp

koox00 commented on components/search_results/search_results.tsx:
```diff
@@ -285,16 +286,14 @@ const SearchResults: React.FC<Props> = (props: Props): JSX.Element => {
             if (searchType === DataSearchTypes.MESSAGES_SEARCH_TYPE && !props.isChannelFiles) {
                 return (
-                    <SearchResultsItem
+                    <PostComponent
```
This will show reactions as well for search items that we didn't show up until now, right?

If that's the case, that reactions are displayed whenever a post is displayed, then we can delete `showDivider` prop in `PostAcknowledgements`

nevyangelova replied:
@koox00 I am not sure about the desired behaviour here. Do you mean post acknowledgements or emoji reactions? @jgilliam17 should we display either of those in search results?

jgilliam17 replied:
@nevyangelova We display acknowledgments in search, pinned, saved and mentions. We do not display post reactions.

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/287457) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/287457)

yasserfaraazkhan commented:
/update-branch

yasserfaraazkhan commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/287495) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/287495)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/287578) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/287578)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/288582) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/288582)

jgilliam17 commented:
Thanks @nevyangelova 
I'll check the e2es over the weekend
Seeing few small issues
1. RHS hijacks focus when clicking on the draft or clicking on pencil icon in global drafts
Precondition:  RHS was open prior to navigating to global drafts
2. Close Edit when changing channel or team (edit: MM-49666 repros on master)
3. RHS: Don't display `update your status` on RHS, only on main channel view, it is causing some crowding and shifting on hover

https://user-images.githubusercontent.com/52937121/212397542-927490e7-58dc-42bc-9ff0-cc194f1bfc8c.mov

4. Tooltip for post actions has small glitch on hover, opens above icon and shows another copy in the top left . Additionally if you click outside of the modal, without dismissing it, post menu remains active
<img width="1482" alt="Screen Shot 2023-01-13 at 2 26 47 PM" src="https://user-images.githubusercontent.com/52937121/212402965-beeb5629-256e-4e28-b4bf-6b40285ad53a.png">

5. Remove emoji reactions to posts on search, saved, mentions and pinned rhs
6. Add back arrow to reply thread when triggering replies on saved, pinned, mentions or search RHS
<img width="418" alt="Screen Shot 2023-01-13 at 4 34 07 PM" src="https://user-images.githubusercontent.com/52937121/212423504-18a34959-3668-42aa-ab72-13649d8e5654.png">

7. Mobile webview: remove Add reaction from post menu on search, pinned, saved and mentions
8. Unable to Upload files via cmd+u or clicking on paperclip icon; dropping files works
9. Keyboard shortcut for cmd+shift+\ for reacting to last post is still broken

m1lt0n commented:
@nevyangelova Taking a look at the latest branch and testing locally, I also see:

1. Clicking on a post in main channel view does not open the thread in RHS
2. After opening the 3 dots post menu, it cannot be closed either clicking on 3 dots or outside the post

nevyangelova commented:
@jgilliam17 some issues are present on community and i am not sure about the behaviour: 

1. Could not reproduce or maybe I don't understand the desired behaviour
2. Lets fix separately
3. Fixed
4. Could not reproduce but I will try again with different server config or npm versions
5.  Could not reproduce, reactions are not present
6. Could not reproduce, arrow is present
7. The menu item is present on community and in the code its stated explicitly to be there. Should we remove it anyway? 
8. Could not reproduce, file upload works in all ways
9. Will fix separately

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/289560) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/289560)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/290854) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/290854)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/292361) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/292361)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/292569) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/292569)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/292994) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/292994)

nevyangelova commented on components/new_post/post_component.tsx:
```diff
@@ -0,0 +1,608 @@
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const isSearchResultItem = (props.matches && props.matches.length > 0) || props.isMentionSearch || (props.term && props.term.length > 0);
```
@koox00 tested this change but for some issues like with showing the thread header, the location will change but its still a search result item, so that is needed for now :/

nevyangelova commented:
/e2e-test

mattermost-build commented:
A pipeline with the same environment variables is already running. 
 Please cancel it first with /e2e-cancel, or specify different environment variables.

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/293005) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/293005)

nevyangelova commented:
@jgilliam17 this is ready for QA review. Regarding the bugs:

@nevyangelova I'll try to add more info and match the original numbering from my previous comment

> 1.RHS hijacks focus when clicking on the draft or clicking on pencil icon in global drafts
This one is not related to the post components logic and is present on community. 

> 3.Tooltip for post actions has small glitch on hover, opens above icon and shows another copy in the top left . 
Fixed.
> 4.Remove emoji reactions to posts on search, saved, mentions and pinned rhs
Fixed.

>5.Add back arrow to reply thread when triggering replies on saved, pinned, mentions or search RHS
Fixed.

> 6.Unable to Upload files via cmd+u or clicking on paperclip icon; dropping files works
This works for me

> 7.Keyboard shortcut for cmd+shift+\ for reacting to last post is still broken
Fixed.

jgilliam17 commented:
/update-branch

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/293939) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/293939)

jgilliam17 commented:
Thanks @nevyangelova 
I checked [e2es](https://mattermost-cypress-report.s3.amazonaws.com/293005-6d28cf7-onprem-ent-webapp-pr-11127-mm-ee-test:6d28cf7/mochawesome.html), and Forward Message tests are failing.
Added ticket for rhs focus when opening drafts - MM-49997
Retested just 2 remaining issues 
1. Keyboard shortcut for cmd+shift+\ for reacting to last post is not working in RHS
2. Back to search arrow is missing when using reply to link on the RHS. See video.

https://user-images.githubusercontent.com/52937121/214942672-e99802a0-56c3-4bb6-b4a1-d4218e340f93.mov

jgilliam17 commented:
[New e2e report](https://mattermost-cypress-report.s3.amazonaws.com/293939-ffd77c0-onprem-ent-webapp-pr-11127-mm-ee-test:ffd77c0/mochawesome.html)
 similar forward message failures, from screenshots, the menu option stays hidden 
This is not failing on [master](https://mattermost-cypress-report.s3.amazonaws.com/293600-onprem-ent-master-mm-ee-test:test/mochawesome.html)
Additionally, channel routing tests are also failing, not sure why the cypress server is displaying different error text, than the one listed in the test - this is not changed in this PR

nevyangelova commented:
@jgilliam17 Fixed the back arrow on clicking reply on thread, the shortcut in RHS will only work on the last post. If the last post is only visible in center, it will emit the shortcut from there.

I saw the forward spec failures but they seem to be only a visual bug in the e2e tests. the behaviour itself work. Can we file a ticket to fix that test separately so we can get this over the finish line?

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/293972) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/293972)

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/294697) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/294697)

nevyangelova commented:
/e2e-test

mattermost-build commented:
The command /e2e-test requires all PR checks to pass.

nevyangelova commented:
/update-branch

nevyangelova commented:
/update-branch

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/295317) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/295317)

nevyangelova commented:
@jgilliam17 fixed the remaining bugs. The closing of the right hand side on threads is happening on community.

nevyangelova commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/295348) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/295348)

jgilliam17 commented:
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/295348-7d7a82c-onprem-ent-webapp-pr-11127-mm-ee-test:7d7a82c/mochawesome.html) 
few failures remain, I will add tickets for follow up after merge.
Otherwise looks good to merge - Thanks @nevyangelova
